### PR TITLE
Dimmer class configuration methods and some additional control methods

### PR DESCRIPTION
**Pull request by PaulWieland**

### CODE REVIEW

File: src/devices/DimmableLightingDevice/DimmableLightingDevice.ts [hunk 1]
```diff
@@ -125,6 +69,7 @@ export default class DimmableLightingDevice extends InsteonDevice {
 					}
 					break;
 				case 0x18: this.emitRemote(['dim','stoppedChanging'], data); break;
+				case 0x21: this.emitRemote(['switch','instant'], data); break;
```
Comment (PaulWieland) [inline src/devices/DimmableLightingDevice/DimmableLightingDevice.ts]: Not sure if this should be `'switch','instant'` or `'switch','on'` or `'switch','on','instant'`. I'm not sure if you would ever want to subscribe to just the instant event.

File: src/devices/DimmableLightingDevice/DimmableLightingDevice.ts [hunk 2]
```diff
@@ -136,24 +81,224 @@ export default class DimmableLightingDevice extends InsteonDevice {
...
 
-	public async getDeviceStatus(){
+    public async getDeviceStatus(){
 
 		// Getting status
 		const statusPacket = await this.statusRequest();
 
 		// Parsing status out
 		return {
+			// state: statusPacket.cmd === 0x00 ? "off" : "on",
```
Comment (PaulWieland) [inline src/devices/DimmableLightingDevice/DimmableLightingDevice.ts]: Was contemplating adding the state `on|off` in addition to the level.

File: src/devices/DimmableLightingDevice/DimmableLightingDevice.ts [hunk 3]
```diff
@@ -216,6 +361,38 @@ export default class DimmableLightingDevice extends InsteonDevice {
 		return this.sendInsteonCommand(0x18,0x00);
 	}
 	
+	/* Not yet tested. Based soley on documentation */
```
Comment (PaulWieland) [inline src/devices/DimmableLightingDevice/DimmableLightingDevice.ts:363]: Will test all three of these as soon as possible - I believe the last batch of dimmer switches that I bought support the light on at rate commands.

File: src/devices/DimmableLightingDevice/DimmableLightingDevice.ts (hunk 1 above)
Comment (apearson) [reply src/devices/DimmableLightingDevice/DimmableLightingDevice.ts]: `'switch','on','instant'`

With eventemitter if you sub to `'switch', 'on'` it should subscribe you to everything under it as well

File: src/devices/DimmableLightingDevice/DimmableLightingDevice.ts (hunk 2 above)
Comment (apearson) [reply src/devices/DimmableLightingDevice/DimmableLightingDevice.ts]: I think just the level would be fine.  The user can do the calculation on their own

File: src/devices/InsteonDevice.ts [hunk 4]
```diff
@@ -446,7 +446,7 @@ export default class InsteonDevice extends EventEmitter2 {
 			            : 'U';
 
 			if(this.options.debug)
-				console.log(`[←][${this.addressString}][${pType}][${data.Flags.Subtype}]: ${toHex(data.cmd1)} ${toHex(data.cmd2)} ${(data.extendedData || []).map(toHex)}`);
```
Comment (apearson) [inline src/devices/InsteonDevice.ts:449]: Let's not change the debug statement

File: src/devices/DimmableLightingDevice/DimmableLightingDevice.ts (hunk 3 above)
Comment (apearson) [reply src/devices/DimmableLightingDevice/DimmableLightingDevice.ts:363]: I can test these on my dimmers if needed

**Review by apearson — CHANGES_REQUESTED** [review]
I've left some comments with changes I'd like.  Let me know if any changes need more discussion

File: src/devices/DimmableLightingDevice/DimmableLightingDevice.ts (hunk 1 above)
Comment (PaulWieland) [reply src/devices/DimmableLightingDevice/DimmableLightingDevice.ts]: Events renamed

**Review by apearson — APPROVED** [review]

**Pull request merged**